### PR TITLE
[Presto-on-Spark] Move DetachedNativeExecutionProcess from test to src

### PR DESCRIPTION
**Pull request by shrinidhijoshi**
Currently DetachedNativeExecutionProcess class is defined in test folder as it was only being used for test execution. But now that we want to support Presto-on-Spark in localMode, we want to be able to talk to an already running CPP process during e2e query execution. Moving it out into src to enable this

Test plan (N/A)
--------------
- Ensure all current tests that use DetachedNativeExecutionProcess pass 

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java [hunk 1]
```diff
@@ -44,6 +44,11 @@ public void configure(Binder binder)
                     config.setMaxConnectionsPerServer(250);
                 });
         binder.bind(PrestoSparkWorkerProperty.class).in(Scopes.SINGLETON);
-        binder.bind(NativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
+        if (System.getProperty("NATIVE_PORT") != null) {
```
Comment (miaoever) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java:47]: should we remove the redundant check at https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/spark/AbstractTestPrestoSparkQueries.java#L137 ?

Comment (shrinidhijoshi) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java:47]: Good catch @miaoever . Looks like there are a few things we can clean up as a part of this. Will update

**Comment by tanjialiang** [comment]
High level question: What is the setup for local mode and why do we need a detached process for local mode? Thanks!

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java (hunk 1 above)
Comment (tanjialiang) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java:47]: Not sure what line it is as that class is no longer there. But things have changed recently so we should double check. Thanks

**Comment by shrinidhijoshi** [comment]
There 2 types of localMode 
1. With standalone cpp process (on devserver)(This PR) - This is needed when you want to start the CPP process in debug mode with breakpoints.

2. With inbuilt cpp process (downloaded through fbpkg) mimicing production - This one is useful when you just want to debug the java side of the flow and do not care about setting up cpp instance on your devserver

**Review by miaoever — APPROVED** [review]
Looks good to me. Thanks for adding this @shrinidhijoshi

**Comment by shrinidhijoshi** [comment]
Resolved merge conflicts

**Comment by shrinidhijoshi** [comment]
Test failures are unrelated. I re-ran the tests on laptop and they pass

**Review by pranjalssh — APPROVED** [review]

**Review by mbasmanova — APPROVED** [review]
@shrinidhijoshi Please, make sure all CI is green.

**Comment by shrinidhijoshi** [comment]
@mbasmanova Will do. Kicked off the failing tests again

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/TestNativeExecutionModule.java
Comment (ajaygeorge) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/TestNativeExecutionModule.java:1]: I understood the rationale for moving code from test folder to src. Why are we deleting this test though?. Is it no longer needed?.

Comment (shrinidhijoshi) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/execution/TestNativeExecutionModule.java:1]: The deleted module didn't have any tests. It only performed the conditional inclusion of DetachedNativeExecution module.
This conditional inclusion has now been moved up into the non-test `src` code so this particular file is no longer needed

**Review by ajaygeorge — APPROVED** [review]
LGTM

**Pull request merged**